src/signer/src/main/java/ee/ria/xroad/signer/tokenmanager/token/SoftwareTokenWorker.java

```diff
@@ -197,6 +195,7 @@ protected void deleteCert(String certId) throws Exception {
         Signature signature = Signature.getInstance(SIGNATURE_ALGORITHM);
```

Comment:
It seems that the algorithm is already checked in SignerUtil.createDataToSign, which also adds a prefix to the digest. Therefore the digest length is longer that expected by the checks. It also means that the additional check is redundant. Maybe there should be a comment explaining why the parameter can be ignored in this case.